### PR TITLE
Rebuilding ddu to drop python-35

### PR DESCRIPTION
**Pull request by Toasterson**
This currently gets me a dependency issue with python that I can't get to go away. For some reason pkgdepend wants to depend on python-35 things even though REQUIRED_PACKAGES correctly add python-37 and 39 only. Can somebody help or test and see if it works on their machines?

### CODE REVIEW

**Comment by mtelka** [comment]
You need to make sure there is no `python3.5` shebang in DDU.  You need to deal with these files:
```
ddu/ddu-text/ddu-text.py
ddu/ddu/ddu.py
ddu/scripts/find_driver.py
ddu/scripts/p5i_install.py
ddu/scripts/url_down.py
```

**Comment by mtelka** [comment]
> You need to make sure there is no `python3.5` shebang in DDU. You need to deal with these files:
> 
> ```
> ddu/ddu-text/ddu-text.py
> ddu/ddu/ddu.py
> ddu/scripts/find_driver.py
> ddu/scripts/p5i_install.py
> ddu/scripts/url_down.py
> ```

... and also all others to make sure they work with newer Python.

**Comment by Toasterson** [comment]
@mtelka I got as far as modifying the ddu sources and removing python-35 but now i get this here

```

/vagrant/components/openindiana/ddu/build/manifest-i386-ddu-text.depend has unresolved dependency '
    depend type=require fmri=__TBD pkg.debug.depend.reason=usr/ddu/ddu-text.py \
        pkg.debug.depend.type=python \
        pkg.debug.depend.file=64/ddu_function.abi3.so \
        pkg.debug.depend.file=64/ddu_function.cpython-39.so \
        pkg.debug.depend.file=64/ddu_function.so \
        pkg.debug.depend.file=ddu_function.abi3.so \
        pkg.debug.depend.file=ddu_function.cpython-39.so \
        pkg.debug.depend.file=ddu_function.py \
        pkg.debug.depend.file=ddu_function.pyc \
        pkg.debug.depend.file=ddu_function.pyo \
        pkg.debug.depend.file=ddu_function.so \
        pkg.debug.depend.file=ddu_function/__init__.py \
        pkg.debug.depend.path=usr/ddu/DDU \
        pkg.debug.depend.path=usr/ddu/ddu-text/DDU \
        pkg.debug.depend.path=usr/lib/python3.9/DDU \
        pkg.debug.depend.path=usr/lib/python3.9/lib-dynload/DDU \
        pkg.debug.depend.path=usr/lib/python3.9/site-packages/DDU \
        pkg.debug.depend.path=usr/lib/python3.9/vendor-packages/DDU \
        pkg.debug.depend.path=usr/lib/python39.zip/DDU'.
/vagrant/components/openindiana/ddu/build/manifest-i386-ddu-text.depend has unresolved dependency '
    depend type=require fmri=__TBD pkg.debug.depend.file=DDU/__init__.py \
        pkg.debug.depend.reason=usr/ddu/ddu-text.py \
        pkg.debug.depend.type=python \
        pkg.debug.depend.path=usr/ddu \
        pkg.debug.depend.path=usr/ddu/ddu-text \
        pkg.debug.depend.path=usr/lib/python3.9 \
        pkg.debug.depend.path=usr/lib/python3.9/lib-dynload \
        pkg.debug.depend.path=usr/lib/python3.9/site-packages \
        pkg.debug.depend.path=usr/lib/python3.9/vendor-packages \
        pkg.debug.depend.path=usr/lib/python39.zip'.
/vagrant/components/openindiana/ddu/build/manifest-i386-ddu.depend has unresolved dependency '
    depend type=require fmri=__TBD pkg.debug.depend.reason=usr/ddu/ddu.py \
        pkg.debug.depend.type=python \
        pkg.debug.depend.file=64/ddu_errors.abi3.so \
        pkg.debug.depend.file=64/ddu_errors.cpython-39.so \
        pkg.debug.depend.file=64/ddu_errors.so \
        pkg.debug.depend.file=ddu_errors.abi3.so \
        pkg.debug.depend.file=ddu_errors.cpython-39.so \
        pkg.debug.depend.file=ddu_errors.py \
        pkg.debug.depend.file=ddu_errors.pyc \
        pkg.debug.depend.file=ddu_errors.pyo \
        pkg.debug.depend.file=ddu_errors.so \
        pkg.debug.depend.file=ddu_errors/__init__.py \
        pkg.debug.depend.path=usr/ddu/DDU \
        pkg.debug.depend.path=usr/lib/python3.9/DDU \
        pkg.debug.depend.path=usr/lib/python3.9/lib-dynload/DDU \
        pkg.debug.depend.path=usr/lib/python3.9/site-packages/DDU \
        pkg.debug.depend.path=usr/lib/python3.9/vendor-packages/DDU \
        pkg.debug.depend.path=usr/lib/python39.zip/DDU'.
/vagrant/components/openindiana/ddu/build/manifest-i386-ddu.depend has unresolved dependency '
    depend type=require fmri=__TBD pkg.debug.depend.reason=usr/ddu/ddu.py \
        pkg.debug.depend.type=python \
        pkg.debug.depend.file=64/ddu_function.abi3.so \
        pkg.debug.depend.file=64/ddu_function.cpython-39.so \
        pkg.debug.depend.file=64/ddu_function.so \
        pkg.debug.depend.file=ddu_function.abi3.so \
        pkg.debug.depend.file=ddu_function.cpython-39.so \
        pkg.debug.depend.file=ddu_function.py \
        pkg.debug.depend.file=ddu_function.pyc \
        pkg.debug.depend.file=ddu_function.pyo \
        pkg.debug.depend.file=ddu_function.so \
        pkg.debug.depend.file=ddu_function/__init__.py \
        pkg.debug.depend.path=usr/ddu/DDU \
        pkg.debug.depend.path=usr/lib/python3.9/DDU \
        pkg.debug.depend.path=usr/lib/python3.9/lib-dynload/DDU \
        pkg.debug.depend.path=usr/lib/python3.9/site-packages/DDU \
        pkg.debug.depend.path=usr/lib/python3.9/vendor-packages/DDU \
        pkg.debug.depend.path=usr/lib/python39.zip/DDU'.
/vagrant/components/openindiana/ddu/build/manifest-i386-ddu.depend has unresolved dependency '
    depend type=require fmri=__TBD pkg.debug.depend.reason=usr/ddu/ddu.py \
        pkg.debug.depend.type=python \
        pkg.debug.depend.file=64/ddu_package.abi3.so \
        pkg.debug.depend.file=64/ddu_package.cpython-39.so \
        pkg.debug.depend.file=64/ddu_package.so \
        pkg.debug.depend.file=ddu_package.abi3.so \
        pkg.debug.depend.file=ddu_package.cpython-39.so \
        pkg.debug.depend.file=ddu_package.py \
        pkg.debug.depend.file=ddu_package.pyc \
        pkg.debug.depend.file=ddu_package.pyo \
        pkg.debug.depend.file=ddu_package.so \
        pkg.debug.depend.file=ddu_package/__init__.py \
        pkg.debug.depend.path=usr/ddu/DDU \
        pkg.debug.depend.path=usr/lib/python3.9/DDU \
        pkg.debug.depend.path=usr/lib/python3.9/lib-dynload/DDU \
        pkg.debug.depend.path=usr/lib/python3.9/site-packages/DDU \
        pkg.debug.depend.path=usr/lib/python3.9/vendor-packages/DDU \
        pkg.debug.depend.path=usr/lib/python39.zip/DDU'.
/vagrant/components/openindiana/ddu/build/manifest-i386-ddu.depend has unresolved dependency '
    depend type=require fmri=__TBD pkg.debug.depend.file=DDU/__init__.py \
        pkg.debug.depend.reason=usr/ddu/ddu.py pkg.debug.depend.type=python \
        pkg.debug.depend.path=usr/ddu \
        pkg.debug.depend.path=usr/lib/python3.9 \
        pkg.debug.depend.path=usr/lib/python3.9/lib-dynload \
        pkg.debug.depend.path=usr/lib/python3.9/site-packages \
        pkg.debug.depend.path=usr/lib/python3.9/vendor-packages \
        pkg.debug.depend.path=usr/lib/python39.zip'.
make[1]: *** [/vagrant/make-rules/ips.mk:538: /vagrant/components/openindiana/ddu/build/.resolved-i386] Error 1
make[1]: Leaving directory '/vagrant/components/openindiana/ddu'
gmake: *** [/vagrant/make-rules/ips.mk:559: REQUIRED_PACKAGES] Error 2
```

How can I solve this?

**Comment by mtelka** [comment]
@Toasterson, what is in `components/openindiana/ddu/build/prototype/i386/usr/lib` directory after you try the publish?  There should be no `python3.5` dir, only `python3.9`.  If there still is `python3.5`, then you still build/install for Python 3.5, not 3.9.

**Comment by Toasterson** [comment]
There is no python-3.5 there. I seem to have problems with autopyc.

**Comment by Toasterson** [comment]
Fixed all the problems by making the ddu makefiles precompile all python files

File: components/openindiana/ddu/Makefile [hunk 1]
```diff
@@ -30,7 +31,7 @@ COMPONENT_REVISION=$(shell cd $(COMPONENT_SRC); git rev-list HEAD --count)
 COMPONENT_SUMMARY=	Device Driver Utility (DDU)
 
 GIT=git
-GIT_REPO=https://github.com/OpenIndiana/ddu
+GIT_REPO=https://github.com/openindiana/ddu
```
Comment (mtelka) [inline components/openindiana/ddu/Makefile]: Why this change?  Organization name at github is OpenIndiana.

File: components/openindiana/ddu/ddu-library.p5m [hunk 2]
```diff
@@ -22,11 +22,11 @@ set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
 license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
 
 <transform file path=usr/ddu/scripts/.+ -> default mode 0555>
-<transform file path=usr/lib/python3.5/vendor-packages/DDU/.*\.py$ \
-	-> default pkg.tmp.autopyc false>
 
 depend type=require fmri=system/library/storage/libmpapi
 
+<include autopyc>
```
Comment (mtelka) [inline components/openindiana/ddu/ddu-library.p5m]: Why is this needed?  `autopyc` should be added automatically - see `ips.mk`.

File: components/openindiana/ddu/ddu-library.p5m [hunk 3]
```diff
@@ -51,9 +51,9 @@ file path=usr/ddu/scripts/pkg_relate.sh
 file path=usr/ddu/scripts/probe.sh
 file path=usr/ddu/scripts/url_down.py
 file path=usr/ddu/scripts/wireless_NIC_keywords
-file path=usr/lib/python3.5/vendor-packages/DDU/__init__.py
-file path=usr/lib/python3.5/vendor-packages/DDU/ddu_devdata.py
-file path=usr/lib/python3.5/vendor-packages/DDU/ddu_errors.py
-file path=usr/lib/python3.5/vendor-packages/DDU/ddu_function.py
-file path=usr/lib/python3.5/vendor-packages/DDU/ddu_package.py
-file path=usr/lib/python3.5/vendor-packages/DDU/ddu_repo.py
+file path=usr/lib/python3.9/vendor-packages/DDU/__init__.py
```
Comment (mtelka) [inline components/openindiana/ddu/ddu-library.p5m]: I suggest to replace `3.9` by `$(PYVER)` so we do not need to change this file on every Python upgrade.

File: components/openindiana/ddu/Makefile (hunk 1 above)
Comment (Toasterson) [reply components/openindiana/ddu/Makefile]: This is an artifact of me using my own repo for the Source changes in DDU and then switching it back but all lowercase since it don't remember the casing offhand.

File: components/openindiana/ddu/ddu-library.p5m (hunk 2 above)
Comment (Toasterson) [reply components/openindiana/ddu/ddu-library.p5m]: Oh, I seem to have lost my change from inside the vagrant VM...

File: components/openindiana/ddu/Makefile (hunk 1 above)
Comment (Toasterson) [reply components/openindiana/ddu/Makefile]: Done

File: components/openindiana/ddu/ddu-library.p5m (hunk 2 above)
Comment (Toasterson) [reply components/openindiana/ddu/ddu-library.p5m]: Done

File: components/openindiana/ddu/ddu-library.p5m (hunk 3 above)
Comment (Toasterson) [reply components/openindiana/ddu/ddu-library.p5m]: Done

**Review by mtelka — APPROVED** [review]
